### PR TITLE
Team header: add/invite widget

### PR DESCRIPTION
**Pull request by jakob223**
cc @keybase/y2ksquad 

I added a second feature flag since we're not going to release the invite stuff at the same time necessarily.

### CODE REVIEW

File: shared/teams/team/nav-header/index.new.tsx [hunk 1]
```diff
@@ -154,6 +181,41 @@ const _HeaderTitle = (props: HeaderTitleProps) => {
...
+          />
+        ) : (
+          <Kb.Box2 direction="vertical" gap="xtiny" alignItems="flex-start">
+            <Kb.CopyText text="https://keybase.io/team/link/blahblah/" />
+            <Kb.Text type="BodyTiny">Adds as writer • Expires 10,000 ys</Kb.Text>
+            <Kb.Text type="BodyTiny" onClick={props.onManageInvites} className="hover-underline">
+              Manage invite links
+            </Kb.Text>
+          </Kb.Box2>
+        ))}
```
Comment (thebearjew) [inline shared/teams/team/nav-header/index.new.tsx]: Since these are siblings why not just wrap in one `flags.teamInvites && ( ... )`?

Comment (thebearjew) [reply shared/teams/team/nav-header/index.new.tsx]: @jakob223 Pointed out that `gap` doesn't work with a Fragment, which is correct. So we would need to update how `Box2` handles `gap` to traverse into Fragments and insert the spacers between.

File: shared/teams/team/nav-header/index.new.tsx [hunk 2]
```diff
@@ -206,6 +279,32 @@ export const SubHeader = (props: SubHeaderProps) =>
...
   () =>
     ({
+      addInviteAndLinkBox: Styles.platformStyles({
+        common: {
+          flexShrink: 0,
+          padding: Styles.globalMargins.tiny,
+        },
+        isElectron: {
+          borderRadius: 4,
+          boxShadow: `0 2px 10px 0 ${Styles.globalColors.black_20}`,
```
Comment (buoyad) [inline shared/teams/team/nav-header/index.new.tsx]: `...Styles.desktopStyles.boxShadow`

Comment (buoyad) [reply shared/teams/team/nav-header/index.new.tsx]: Actually I think you could do this in `desktop/renderer/style.css` to make the hover behavior in the design work.

Comment (jakob223) [reply shared/teams/team/nav-header/index.new.tsx]: Whoa, I totally missed that that was what was going on with this drop shadow situation

File: shared/desktop/renderer/style.css [hunk 3]
```diff
@@ -439,3 +439,11 @@ body {
 .darkMode .emoji-mart-bar {
   border-color: rgba(0, 0, 0, 0.85);
 }
+
+.headerTitle:hover .addInviteAndLinkBox {
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
+}
+
+.darkMode .headerTitle:hover .addInviteAndLinkBox {
```
Comment (buoyad) [inline shared/desktop/renderer/style.css]: Would be snazzy to also have a `transition` on this.

**Pull request merged**